### PR TITLE
[DXCDT-110] [DXCDT-113] Refactor signing key, prompt, hook, job tests

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

Refactor signing key, prompt, hook, job tests so they can be ran in any order and they clean up the test tenant after running.

## Testing

- [x] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used, if not `main`.

### CODE REVIEW

File: management/signing_key_test.go [hunk 1]
```diff
@@ -2,70 +2,56 @@ package management
...
-		}
+		signingKeys, err := m.SigningKey.List()
+		assert.NoError(t, err)
+		assert.NotEmpty(t, signingKeys, "expected at least one key to be returned")
 
 		// save kid for later use
-		kid = ks[0].GetKID()
-
-		t.Logf("%v\n", ks)
+		kid = signingKeys[0].GetKID()
```
Comment (Widcket) [inline management/signing_key_test.go:18]: Could these tests run in any order?

**Review by Widcket — CHANGES_REQUESTED** [review]
Good work! Left a question.

File: management/signing_key_test.go (hunk 1 above)
Comment (sergiughf) [reply management/signing_key_test.go]: Good catch! Thanks for spotting this:) I somehow missed it. Fixed in https://github.com/auth0/go-auth0/pull/60/commits/9c910cdb0b3bf78e60a012783756a36cfb0985a1.

**Review by Widcket — APPROVED** [review]
LGTM

**Pull request merged**